src/app/item-page/edit-item-page/item-bitstreams/item-bitstreams.service.ts

```diff
@@ -0,0 +1,569 @@
+   * selected bitstream already is at the beginning of the bundle.
+   */
+  moveSelectedBitstreamUp() {
+    const selected = this.getSelectedBitstream();
+
+    if (hasNoValue(selected) || this.getPerformingMoveRequest()) {
+      return;
+    }
+
+    const originalPosition = selected.currentPosition;
+    if (originalPosition > 0) {
+      const newPosition = originalPosition - 1;
+      selected.currentPosition = newPosition;
+
+      const onRequestCompleted = () => {
+        this.announceMove(selected.bitstream.name, newPosition);
+      };
+
+      this.performBitstreamMoveRequest(selected.bundle, originalPosition, newPosition, onRequestCompleted);
+      this.updateSelectionAction({ action: 'Moved', selectedEntry: selected });
+    }
+  }
+
+  /**
+   * Moves the selected bitstream one position down in the bundle. Does nothing if no bitstream is selected or the
+   * selected bitstream already is at the end of the bundle.
+   */
+  moveSelectedBitstreamDown() {
+    const selected = this.getSelectedBitstream();
+
+    if (hasNoValue(selected) || this.getPerformingMoveRequest()) {
+      return;
+    }
+
+    const originalPosition = selected.currentPosition;
+    if (originalPosition < selected.bundleSize - 1) {
+      const newPosition = originalPosition + 1;
+      selected.currentPosition = newPosition;
+
+      const onRequestCompleted = () => {
+        this.announceMove(selected.bitstream.name, newPosition);
+      };
+
+      this.performBitstreamMoveRequest(selected.bundle, originalPosition, newPosition, onRequestCompleted);
+      this.updateSelectionAction({ action: 'Moved', selectedEntry: selected });
+    }
+  }
+
+  /**
+   * Sends out a Move Patch request to the REST API, display notifications,
+   * refresh the bundle's cache (so the lists can properly reload)
+   * @param bundle The bundle to send patch requests to
+   * @param fromIndex The index to move from
+   * @param toIndex The index to move to
+   * @param finish Optional: Function to execute once the response has been received
+   */
+  performBitstreamMoveRequest(bundle: Bundle, fromIndex: number, toIndex: number, finish?: () => void) {
+    if (this.getPerformingMoveRequest()) {
+      console.warn('Attempted to perform move request while previous request has not completed yet');
+      return;
+    }
+
+    const moveOperation: MoveOperation = {
+      op: 'move',
+      from: `/_links/bitstreams/${fromIndex}/href`,
+      path: `/_links/bitstreams/${toIndex}/href`,
+    };
+
+    this.announceLoading();
+    this.isPerformingMoveRequest.next(true);
+    this.bundleService.patch(bundle, [moveOperation]).pipe(
+      getFirstCompletedRemoteData(),
+      tap((response: RemoteData<Bundle>) => this.displayFailedResponseNotifications(MOVE_KEY, [response])),
+      switchMap(() => this.requestService.setStaleByHrefSubstring(bundle.self)),
+      take(1),
+    ).subscribe(() => {
+      console.log('got here!');
```

Comment:
Please remove this console.log.  It's obviously this was for testing purposes. :)

Reply:
Ah, good catch! 😅 I've updated the 3 PRs.